config/components/rbac/kustomization.yaml

```diff
@@ -9,12 +9,15 @@ resources:
 - role_binding.yaml
 - leader_election_role.yaml
 - leader_election_role_binding.yaml
-# Comment the following 3 lines if you want to disable
-# the auth proxy (https://github.com/brancz/kube-rbac-proxy)
-# which protects your /metrics endpoint.
-- auth_proxy_role.yaml
-- auth_proxy_role_binding.yaml
-- auth_proxy_client_clusterrole.yaml
+# The following RBAC configurations are used to protect
+# the metrics endpoint with authn/authz. These configurations
+# ensure that only authorized users and service accounts
+# can access the metrics endpoint. Comment the following
+# permissions if you want to disable this protection.
+# More info: https://book.kubebuilder.io/reference/metrics.html
+- metrics_auth_role.yaml
+- metrics_auth_role_binding.yaml
+- metrics_reader_role.yaml
```

Comment:
BTW I just figured out what I needed in jobset to get this to work. I think you need a binding role for metrics_reader_role.yaml.

https://github.com/kubernetes-sigs/jobset/pull/722/files#diff-152cfa6c5cac6cad739a2959d2626f824cc2ece7a8bdc8243a9e5759e407726b

Reply:
Yes, this helps. Thank you so much! I also checked with this https://github.com/kubernetes-sigs/jobset/pull/722#issuecomment-2523687506, and it's working.

Reply:
Actually, this is disabled by default in Kubebuilder. Should we enable it?

Reply:
It depends on what we want to assume. I expect prod users would configure Prometheus and cert manager. This gets it working with default installation. If we don’t include it then metrics reader can’t access the endpoint.

Reply:
IIUC cert-manager cannot be actually used OOTB because we don't set the CertDir, and other parameters: https://github.com/kubernetes-sigs/kubebuilder/blob/e07823e33190b4d3b9339a8273183187b9b50535/testdata/project-v4/cmd/main.go#L129-L133. I'm wondering if we should open an issue to support that, or reuse this issue: https://github.com/kubernetes-sigs/kueue/issues/3259.

Reply:
This parameters set by default 

https://github.com/kubernetes-sigs/controller-runtime/blob/main/pkg/metrics/server/server.go#L85-L102

Reply:
Interesting, where is the defaulting done?  Does it work in the way we integrate with this? The comment I linked suggests the lines need to be uncommented for the params to be effective.